internal/cmd/gocdk/serve.go

```diff
@@ -22,19 +22,25 @@ import (
 	"log"
 	"net"
 	"net/http"
+	"net/http/httputil"
 	"net/url"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
+	"sync/atomic"
 	"time"
 
+	"golang.org/x/sync/errgroup"
 	"golang.org/x/xerrors"
 )
 
 func serve(ctx context.Context, pctx *processContext, args []string) error {
 	f := newFlagSet(pctx, "serve")
-	address := f.String("address", "localhost:8080", "address to serve on")
-	biome := f.String("biome", "dev", "biome to apply and use configuration from")
+	opts := new(serveOptions)
+	f.StringVar(&opts.address, "address", "localhost:8080", "`host:port` to address to serve on")
+	f.StringVar(&opts.biome, "biome", "dev", "`name` of biome to apply and use configuration from")
+	f.IntVar(&opts.basePort, "base-port", 9090, "use `port` and port+1 for the built servers to listen on")
```

Comment:
Consider a flag to disable the reverse proxy and just run the server directly?

Reply:
The user can directly connect to the server at `localhost:base-port` or run the server with `go run`. Configurability of whether to reverse-proxy or not seems like it add more branching to the logic for a relatively uncommon case.

Reply:
Will `go run` actually work? IIUC, `gocdk serve` is doing some work other than just launching the server, no? E.g., generating the config file, adding a command-line flag with the config.

I can think of a number of reasons why this "run 2 servers proxying 1 of them" thing might not work for a given application:

* It binds to a specific port other than the HTTP one (e.g., gRPC server).
* It connects to some kind of global resource (e.g., subscribes to a topic).
* It writes to a specific file on the filesystem.

It's fine with me to add a TODO for this, but I think we need a reasonable escape hatch. If `go run` works, that would be fine, but I'm not sure it will. Connecting to the base-port directly doesn't help.

Reply:
Opened #1914 to track.

Reply:
@vangent if a user doesn't desire the live reloading, or it doesn't work for their application for any reason, in terms of escape hatch, we've said that they will also be able to deploy to local (and directly launch to local, as long as they've built the docker image at least once).